src/main/scala/com/fulcrumgenomics/personal/nhomer/HasSequence.scala

```diff
@@ -0,0 +1,157 @@
+  *
+  * 2. Redistributions in binary form must reproduce the above copyright notice,
+  * this list of conditions and the following disclaimer in the documentation
+  * and/or other materials provided with the distribution.
+  *
+  * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+  * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+  * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+  * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+  * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+  * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+  * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+  * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+  * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+  * POSSIBILITY OF SUCH DAMAGE.
+  */
+
+package com.fulcrumgenomics.personal.nhomer
+
+import java.nio.file.Path
+
+import com.fulcrumgenomics.cmdline.{JeanLucTool, Personal}
+import dagr.commons.CommonsDef.PathToBam
+import dagr.commons.io.Io
+import dagr.sopt._
+import htsjdk.samtools._
+import htsjdk.samtools.metrics.{MetricBase, MetricsFile}
+import htsjdk.samtools.util._
+import picard.PicardException
+
+import scala.io.Source
+
+@clp(
+  description = "Searches for DNA sequences in the read pairs.",
+  group = classOf[Personal]
+)
+class HasSequence
+( @arg(doc = "Input SAM or BAM.") val input: PathToBam,
+  @arg(doc = "Output SAM or BAM.") val output: PathToBam,
+  @arg(doc = "File containing the DNA sequences to search for.") val sequences: Path,
+  @arg(doc = "Sequence match histogram written to this file.") val metrics: Path,
+  @arg(doc = "Maximum mismatches for matching a sequence.") val maxMismatches: Int = 1,
+  @arg(doc = "Minimum base quality. Any bases falling below this quality will be considered a mismatch even in the bases match.", flag = "q") val minimumBaseQuality: Int = 0,
+  @arg(doc = "Treat no calls in the input sequences as DNA bases.") val noCallsAreBases: Boolean = false,
+  @arg(doc = "Search for mismatches at the start only") val startOnly: Boolean = true,
+  @arg(doc = "The tag to store the result.") val hasSequenceTag: String = "XW"
+) extends JeanLucTool {
+  Io.assertReadable(input)
+  Io.assertCanWriteFile(output)
+  Io.assertCanWriteFile(metrics)
+
+  // TODO: progress logging and metrics file header
+  override def execute: Int = {
+    val sequencesReadOne: Iterator[String] = for(line <- Source.fromFile(this.sequences.toFile).getLines()) yield line.trim
+    val sequencesReadTwo: Iterator[String] = for (sequence <- sequencesReadOne) yield SequenceUtil.reverseComplement(sequence)
+    val reader: SamReader = SamReaderFactory.makeDefault.open(input.toFile)
+    val header: SAMFileHeader = reader.getFileHeader
```

Comment:
And already I want a scala wrapper of htsjdk so we can write `reader.header` without the `get`